In [1]:
import pandas as pd
import numpy as np

In [2]:
dataset = pd.read_csv("PrePlacement.csv")
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       0.0  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       0.0  

[215 rows x 15 columns]

In [3]:
from scipy.stats import ttest_ind
dataset = dataset.dropna()
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       0.0  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       0.0  

[215 rows x 15 columns]

In [6]:
[dataset['gender'] == 'F']

[0      False
 1      False
 2      False
 3      False
 4      False
        ...  
 210    False
 211    False
 212    False
 213     True
 214    False
 Name: gender, Length: 215, dtype: bool]

In [7]:
dataset[dataset['gender'] == 'M']['salary']

0      270000.0
1      200000.0
2      250000.0
3           0.0
4      425000.0
         ...   
209    216000.0
210    400000.0
211    275000.0
212    295000.0
214         0.0
Name: salary, Length: 139, dtype: float64

In [8]:
dataset[dataset['gender'] == 'F']['salary']

6           0.0
12          0.0
13     218000.0
15     200000.0
17          0.0
         ...   
197    260000.0
198         0.0
204    210000.0
208         0.0
213    204000.0
Name: salary, Length: 76, dtype: float64

In [10]:
from scipy.stats import ttest_ind
dataset = dataset.dropna()
male = dataset[dataset['gender'] == 'M']['salary']
female = dataset[dataset['gender'] == 'F']['salary']
ttest_ind(male,female)

TtestResult(statistic=2.0937842507001165, pvalue=0.03746230397067218, df=213.0)

p-value < 0.05 → Significant difference
The positive t-statistic (2.09) suggests Group A > Group B
Insight:
There is a statistically significant difference between the two groups being compared. The first group has a higher mean than the second group.
There is a significant difference between in salaries of Male and Female Candidates.
Male students earn higher salaries when compared to Female students. 

In [ ]:
#Central & Others - Salary

In [12]:
from scipy.stats import ttest_ind
dataset = dataset.dropna()
central = dataset[dataset['ssc_b'] == 'Central']['salary']
others = dataset[dataset['ssc_b'] == 'Others']['salary']
ttest_ind(central,others)

TtestResult(statistic=-0.6182213205902315, pvalue=0.5370899470596882, df=213.0)

p-value > 0.05 → Not significant
The negative t-statistic suggests Group B might have a slightly higher mean,
but the difference is not important.
Insight:
No statistically significant difference between the groups. Any difference is due to chance.
There is no difference of salary between the Central board and Other board students.

Dependent - Paired - T-Test

In [13]:
from scipy.stats import ttest_ind
dataset = dataset.dropna()
male = dataset[dataset['gender'] == 'M']['ssc_p']
male1 = dataset[dataset['gender'] == 'M']['hsc_p']
ttest_ind(male,male1)

TtestResult(statistic=0.4609004016724563, pvalue=0.6452330329980636, df=276.0)

p-value > 0.05 → No significant difference
The t-value is very small (0.46), showing almost no difference.
Insight:
The two groups likely have similar means. No meaningful difference exists.
SSc and HSC Marks are similar. NO difference between them.

In [ ]:
#Commerce - Entrance Test & MBA

In [15]:
from scipy.stats import ttest_ind
dataset = dataset.dropna()
etest = dataset[dataset['hsc_s'] == 'Commerce']['etest_p']
mba = dataset[dataset['hsc_s'] == 'Commerce']['mba_p']
ttest_ind(etest,mba)

TtestResult(statistic=7.079422262550767, pvalue=1.844299454870846e-11, df=224.0)

p-value is extremely small (< 0.000001)
Very strong statistical significance
The very high positive t-statistic (7.08) shows a large difference

Insight:
There is a highly significant and meaningful difference between the two groups.
Group A’s mean is much higher than Group B's.
There is a difference between entrance marks and MBA marks.
But, Entrance Test marks have higher significance.

In [ ]:
#Hypothesis Testing

In [ ]:
#Stetement: Test Whether ssc_p of male and hsc_p of female is same?

In [ ]:
#Null Hypothesis - There is no significance between 10th and 12th marks of male students.

In [16]:
from scipy.stats import ttest_ind
dataset = dataset.dropna()
male = dataset[dataset['gender'] == 'M']['ssc_p']
male1 = dataset[dataset['gender'] == 'M']['hsc_p']
ttest_ind(male,male1)

TtestResult(statistic=0.4609004016724563, pvalue=0.6452330329980636, df=276.0)

In [ ]:
#p<0.05 ---> reject Null Hypothesis
#Here, p>0.05 ---> Accept Null Hypothesis and reject Alternate Hypothesis

ANOVA

In [19]:
# ONE-WAY CLASSIFICATION
from scipy.stats import stats
stats.f_oneway(dataset['ssc_p'],dataset['hsc_p'],dataset['degree_p'])

/var/folders/_d/9dyxj4t14ql55yv2t3fzbr7c0000gn/T/ipykernel_39992/4226218398.py:3: DeprecationWarning: Please import `f_oneway` from the `scipy.stats` namespace; the `scipy.stats.stats` namespace is deprecated and will be removed in SciPy 2.0.0.
  stats.f_oneway(dataset['ssc_p'],dataset['hsc_p'],dataset['degree_p'])


F_onewayResult(statistic=0.695991304348204, pvalue=0.49895574725815317)

p<0.05, Here p>0.05. i.e, p is 0.49.
So, there is a significant difference between School Marks and Degree Marks.

In [20]:
# TWO-WAY CLASSIFICATION
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
from scipy.stats import stats
#Example: Two-way ANOVA
model = ols('salary ~ C(gender) + C(mba_p) + C(gender):C(mba_p)', data=dataset).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)

                          sum_sq     df         F    PR(>F)
C(gender)           9.089736e+08    1.0  0.038193  0.850607
C(mba_p)            5.617482e+12  204.0  1.157035  0.458317
C(gender):C(mba_p)  4.750222e+12  204.0  0.978405  0.581251
Residual            1.665955e+11    7.0       NaN       NaN


/opt/anaconda3/envs/aiml/lib/python3.11/site-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 204, but rank is 45
  warnings.warn('covariance of constraints does not have full '
/opt/anaconda3/envs/aiml/lib/python3.11/site-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 204, but rank is 132
  warnings.warn('covariance of constraints does not have full '


C(gender) → p = 0.8506 → Not significant
C(mba_p) → p = 0.4583 → Not significant
Interaction (gender × mba_p) → p = 0.5812 → Not significant
Since all p-values are greater than 0.05, there is no statistically significant effect from gender, MBA percentage, 
or their interaction on your dependent variable.

In [21]:
# TWO-WAY CLASSIFICATION
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
from scipy.stats import stats
#Example: Two-way ANOVA
model = ols('salary ~ C(gender) + C(etest_p) + C(gender):C(etest_p)', data=dataset).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)

                            sum_sq    df         F    PR(>F)
C(gender)                      NaN   1.0       NaN       NaN
C(etest_p)            1.527889e+12  99.0  0.655749  0.932319
C(gender):C(etest_p)  2.340092e+12  99.0  1.004336  0.490080
Residual              2.047567e+12  87.0       NaN       NaN


/opt/anaconda3/envs/aiml/lib/python3.11/site-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 1, but rank is 0
  warnings.warn('covariance of constraints does not have full '
/opt/anaconda3/envs/aiml/lib/python3.11/site-packages/statsmodels/base/model.py:1923: RuntimeWarning: invalid value encountered in divide
  F /= J
/opt/anaconda3/envs/aiml/lib/python3.11/site-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 99, but rank is 41
  warnings.warn('covariance of constraints does not have full '
/opt/anaconda3/envs/aiml/lib/python3.11/site-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 99, but rank is 75
  warnings.warn('covariance of constraints does not have full '


C(etest_p) → p = 0.9323 → Not significant
Interaction (gender × etest_p) → p = 0.490 → Not significant
Since all p-values are greater than 0.05, there is no statistically significant effect from gender, Entrance percentage, 
or their interaction on your dependent variable.